src/domain/safe-apps/entities/safe-app.entity.ts

```diff
@@ -0,0 +1,14 @@
+import { SafeAppAccessControl } from './safe-app-access-control.entity';
+import { SafeAppProvider } from './safe-app-provider.entity';
+
+export interface SafeApp {
+  id: number;
+  url: string;
+  name: string;
+  iconUrl: string;
+  description: string;
+  chainIds: string[];
```

Comment:
Should be an array of integers. https://github.com/safe-global/safe-config-service/blob/main/src/safe_apps/models.py#L48

Reply:
Correct, this is an array of integers in the data source (Config Service), but the current implementation of the client gateway converts the integers to strings (see [this](https://safe-client.staging.5afe.dev//v1/chains/5/safe-apps)). So by defining this as an array of strings in the AJV schema, in combination with `coerceTypes: true` in `json-schema.service.ts` performs a type conversion when validating. 

I left a comment in `json-schema.service.ts` precisely to document that behaviour, and thought this will be changed when implementing serialisation. This works, and the E2E test is getting an array of strings (because AJV is doing the conversion) so nothing is wrong with E2E in this case.

Since we recently decided to implement ad hoc mappings for serialisation, maybe it's time to remove this `coerceTypes` option from AJV config and implement an ad hoc mapping for this and maybe another coerced types. What do you think? I can do a separate PR for mapping this and other potentially affected types @fmrsabino @moisses89.

Reply:
Understood @hectorgomezv I'd say that we should do but not sure if now. I'd do it in a different PR for sure. :)

Reply:
> Correct, this is an array of integers in the data source (Config Service), but the current implementation of the client gateway converts the integers to strings (see [this](https://safe-client.staging.5afe.dev//v1/chains/5/safe-apps)).

We have two entities in the Safe Client Gateway. The domain entity should follow the spec of the data source. See [here](https://github.com/safe-global/safe-client-gateway/blob/37372e8ad8ecc6cd56860d7a22ac63291ce6b72d/src/common/models/backend/safe_apps.rs#L11).

We then have the route entity where it becomes a `String`.

The Schema validation was done to validate the schema of responses returned by other datasources. In this case it should be `number`.

Reply:
Yes, I agree @fmrsabino , I tried to express the same in my previous message. This type-coercion solution was included in the project first steps when serialisation wasn't a concern. In fact I just tried to remove the `coerceTypes: true` property from AJV configuration and several controller tests fail as result: chains, balances, and this new safe-apps.

I think it's better to solve this in a separate PR to stay single-purpose. I can do it today if you agree 🙂

Reply:
Yeah I agree. Let's open a PR with this change and see what is affected by it 👍